src/sqliteformat.cpp

```diff
@@ -366,6 +474,55 @@ bool SqliteFormat::purgeDeletedComponents(const KCalendarCore::Incidence &incide
     return false;
 }
 
+bool SqliteFormat::purgeAllComponents(const QString &notebook)
+{
+    if (notebook.isEmpty()) {
+        qCWarning(lcMkcal) << "empty notebook uid, cannot purge all components.";
+        return false;
+    }
+
+    int rv = 0;
+    int index = 1;
+    const char *query;
+    int qsize = 0;
+    sqlite3_stmt *stmt = nullptr;
+
+    const QByteArray n = notebook.toUtf8();
+
+    if (!d->mDeleteIncComponents) {
+        const char *query = DELETE_COMPONENTS;
```

Comment:
Not really. I'm passing directly the macro now instead of putting in a variable. And I move the one letter 'n' variable some lines later into a two letters one !